### PR TITLE
feat(plugin): RevealAllSpoilers

### PR DESCRIPTION
**Pull request by whqwert**
Port of RevealAllSpoilers from BD

### CODE REVIEW

File: src/plugins/revealAllSpoilers.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    reveal(event: SpoilerEvent) {
+        const { ctrlKey, shiftKey, target } = event;
+        if (!ctrlKey) { return; }
+
+        const parent = shiftKey
+            ? document.querySelector("div[class*=messagesWrapper-]")
+            : target.parentElement;
+
+        for (const spoiler of parent!.querySelectorAll(
+            "span[class*=spoilerText-][class*=hidden-]"
```
Comment (AAGaming00) [inline src/plugins/revealAllSpoilers.ts:52]: HUSK

**Comment by castdrian** [comment]
? why, this already exists in the client
Discord_uSlg43GC91

File: src/plugins/revealAllSpoilers.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { findByPropsLazy } from "@webpack";
+
+interface SpoilerEvent {
+    ctrlKey: boolean;
+    shiftKey: boolean;
+    target: HTMLSpanElement;
+}
```
Comment (Vendicated) [inline src/plugins/revealAllSpoilers.ts:27]: this is just HTMLMouseEvent https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent

File: src/plugins/revealAllSpoilers.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    name: "RevealAllSpoilers",
+    description: "Reveal all spoilers in a message (Ctrl) or in the chat (Ctrl+Shift) and clicking a spoiler.",
+    authors: [Devs.whqwert],
+
+    patches: [
+        {
+            find: ".revealSpoiler=function",
+            replacement: {
+                match: /\.revealSpoiler=function\((.{1,2})\){/,
+                replace: ".revealSpoiler=function($1){Vencord.Plugins.plugins.RevealAllSpoilers.reveal($1);"
```
Comment (Vendicated) [inline src/plugins/revealAllSpoilers.ts:42]:
```suggestion
                replace: ".revealSpoiler=function($1){$self.reveal($1);"
```

File: src/plugins/revealAllSpoilers.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { findByPropsLazy } from "@webpack";
+
+const SpoilerClasses = findByPropsLazy("spoilerText");
+const MessagesClasses = findByPropsLazy("messagesWrapper", "messages");
+
+export default definePlugin({
+    name: "RevealAllSpoilers",
+    description: "Reveal all spoilers in a message (Ctrl) or in the chat (Ctrl+Shift) and clicking a spoiler.",
```
Comment (Vendicated) [inline src/plugins/revealAllSpoilers.ts:28]: i dont understand this description, could u rephrase it please?

**Comment by Vendicated** [comment]
cool thanks!!

**Pull request merged**